share/p2p/peers/manager.go

```diff
@@ -189,6 +190,8 @@ func (m *Manager) Stop(ctx context.Context) error {
 	select {
 	case <-m.headerSubDone:
 		return nil
+	case <-m.disconnectedPeersDone:
```

Comment:
no we need to wait for both to return.

Reply:
This is a lot more complicated if we do it right (with a wg), is this enough?

```golang
	select {
	case <-m.headerSubDone:
	case <-ctx.Done():
		return ctx.Err()
	}

	select {
	case <-m.disconnectedPeersDone:
	case <-ctx.Done():
		return ctx.Err()
	}
```

Reply:
Why do we necessarily need to wait here? Not immediately obvious.

Reply:
for i 2{
<- chan1
<- chan2
<- ctx.Done
}

Reply:
We have this is HeaderExchange

Reply:
for loop wil just read twice from same channel

Reply:
If you close it but not if you send <- struct{}{}